go-controller/pkg/node/node_ip_handler_linux_test.go

```diff
@@ -152,6 +155,66 @@ var _ = Describe("Node IP Handler tests", func() {
 	})
 })
 
+var _ = Describe("Node IP Handler tests", func() {
+	// To ensure that variables don't leak between parallel Ginkgo specs,
+	// put all test context into a single struct and reference it via
+	// a pointer. The pointer will be different for each spec.
+	var tc *testCtx
+
+	const (
+		nodeName                = "node1"
+		dummyBrName             = "breth0"
+		dummyBrInternalIPv4     = "10.1.1.10"
+		dummyAdditionalIPv4CIDR = "192.168.2.2/32"
+	)
+
+	BeforeEach(func() {
+		fexec := ovntest.NewFakeExec()
+		Expect(util.SetExec(fexec)).ShouldNot(HaveOccurred())
+		fexec.AddFakeCmd(&ovntest.ExpectedCmd{
+			Cmd:    "ovs-vsctl --timeout=15 get Open_vSwitch . external_ids:ovn-encap-ip",
+			Output: dummyBrInternalIPv4,
+		})
+		useNetlink, createNs, createPrimaryLink := true, true, true
+		tc = configureKubeOVNContext(nodeName, useNetlink, createNs, createPrimaryLink)
+		tc.ipManager.syncPeriod = 10 * time.Millisecond
+		tc.doneWg.Add(1)
+		go tc.ns.Do(func(netNS ns.NetNS) error {
+			tc.ipManager.runInternal(tc.stopCh, tc.ipManager.getNetlinkAddrSubFunc(tc.stopCh))
+			tc.doneWg.Done()
+			return nil
+		})
+	})
+
+	AfterEach(func() {
+		close(tc.stopCh)
+		tc.doneWg.Wait()
+		tc.watchFactory.Shutdown()
+		Expect(tc.ns.Close()).ShouldNot(HaveOccurred())
+	})
+
+	Context("valid addresses", func() {
+		It("allows keepalived VIP", func() {
```

Comment:
can you make sure we add a unit test for:
```
link local addresses, OVN reserved IPs, and addresses marked as secondary or deprecated.
```
and double check to see these are not filtered at this stage?
if I am asking for too much and perhaps most of these are in the same line as what you have added, push back please but since you already have all the plumbing in place I was hoping this will be easier

Reply:
Link local addresses would be filtered - but uniq link local addresses are not and I added a test for that.

Reply:
ack thanks! and perhaps also for masquerade IPs